### DISCUSSION TITLE
Disable climate away mode switches device to "Auto" mode

### DISCUSSION THREAD

**Discussion by hcrohland** [SukramJ/aiohomematic#750] [General]
DONT'T DELETE THIS.

**custom_component version (if applicable):**
1.23.0

**known last working custom_component version (if applicable):**
none

**Home Assistant:**
- Version: 2022.12.0
- Installation method (HAOS, supervised, core, venv, Docker):  Docker

**CCU:**
- Model (CCU2/3, Raspi3/4, NUC, ProxmoxVE, HA-Addon ...): Raspi 4
- OS (Original, RaspberryMatic, debmatic ...): RaspberryMatic
- Version: 

**Problem-relevant configuration:**

Do you use tls? no
Do you use callback? no
Do you use username and password? yes 
Which interfaces do you use (hmip/bidcos/wired)? hmip

**Describe the bug**
Calling Service "Disable climate away mode" switches device into "Auto" mode even if they were in "Manu" mode before. 

**To Reproduce**
Call Service "Disable climate away mode" for a device in "Manu" mode.
The device will switch to "Auto" Mode

**Expected behavior**

Disabling Climate away should not change the mode of entities

**Comment by SukramJ** [comment]
Since we don't cache states outside of the devices, I had to make a decision about what happens when manually disabling away mode.
The current operating mode of the thermostat is controlled with the parameter CONTROL_MODE (Auto, Manu, Away,...).

The device does not store which status was previously selected.
So my decision was to jump back into automatic operation, which makes sense to me.

It is not an option to save the last state in HA (we generally don't do that), since Away Mode in particular can be active for a longer period of time, and of course there could be some HA restarts in between.

So if you find a way how the last state before the away mode is stored in the Homematic device, I can of course correct it in the HA code

**Reply by hcrohland** [reply]
> The device does not store which status was previously selected.

Yes, it does.

And 

```
service: homematicip_local.enable_away_mode_by_calendar
data:
  end: "{{ now() }}"
target:
  entity_id:
```

does the trick

**Reply by SukramJ** [reply]
> The device does not store which status was previously selected.

It is clear that the status is stored internally, but since it cannot be called up via the API, it cannot be used as a value either.

**Reply by SukramJ** [reply]
If you already had a solution, it would of course have been interesting in the first post. ;-)

**Reply by hcrohland** [reply]
And you do not need the value. Just set the end time to now...

**Reply by hcrohland** [reply]
I digged into it after opening the post, sorry

**Reply by SukramJ** [reply]
Good job.

**Reply by SukramJ** [reply]
Will be part of next release.

**Reply by hcrohland** [reply]
You should probably do a bit into the future. With now() the target temperature seems to be stuck at the away temp sometimes. 
With + 5 minutes it seems to be fine for me.

**Reply by SukramJ** [reply]
I use `2000_01_01 00:00`

**Reply by hcrohland** [reply]
With that it is setting the temperature to the away temperature. Apparently the thermostat is first resetting to auto and its temperature and then setting the temp for the away mode!?!

**Reply by SukramJ** [reply]
The result is that the away mode is off afterwards, and the device goes back to previous mode.
Whatever steps are made by the device in between.

```
    async def disable_away_mode(self) -> None:
        """Disable the away mode on thermostat."""
        await self.put_paramset(
            paramset_key="VALUES",
            value={
                "CONTROL_MODE": HMIP_MODE_AWAY,
                "PARTY_TIME_START": PARTY_INIT_DATE,
                "PARTY_TIME_END": PARTY_INIT_DATE,
            },
        )
```

**Reply by SukramJ** [reply]
Test it with version 1.24.2.

**Reply by hcrohland** [reply]
Works great!
